config/testutil.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2015 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License. See the AUTHORS file
+// for names of contributors.
+//
+// Author: Marc Berhault (marc@cockroachlabs.com)
+
+package config
+
+import "sync"
+
+var (
+	testingZoneConfig   map[uint32]*ZoneConfig
+	testingHasHook      bool
+	testingPreviousHook func(*SystemConfig, uint32) (*ZoneConfig, error)
+	testingLock         sync.Mutex
+)
+
+// TestingZoneConfigCloser only exists to provide an stopper.Closer.
+type TestingZoneConfigCloser struct{}
+
+// Close implements stopper.Closer.
+func (t *TestingZoneConfigCloser) Close() {
+	TestingResetZoneConfigHook()
+}
+
+// TestingSetupZoneConfigHook initializes the zone config hook
+// to 'testingZoneConfigHook' which uses 'testingZoneConfig'.
+// NOTE: you must call TestingResetZoneConfigHook at the end of your test case.
```

Comment:
Could this function take a `*stop.Stopper` and call `AddCloser` for you, or is it sometimes used where there is no appropriate Stopper?